module-4/backend/week_two.md

```diff
@@ -1,22 +1,59 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
-
-Note: When you're done, submit a PR. 
-
 1. What's one difference between ES5 and ES6?
+
+  Far fewer semi colons in ES6.
+  
 2. What's the difference between asynchronous and synchronous JavaScript? 
+
+  Asynchoronous code is excuted simultaenously while synchronous is done in order one piece at a time. 
+
 3. What are the four pillars of Object Oriented programming?
+
+  Encapsulation, Inheritance, Abstraction, Polymorphism.
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+
+  Promises, constructors, this , classes. 
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+  Inheritance, and exporting. 
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+
+  A callback function is a function thats passed into another. 
+  
 7. What's the scope of variables in Javascript?
-8. What's the difference between `==` and `===` in JavaScript?
+
+  Local
+  
+8. What's the difference between `==` and `===` in JavaScript?\
+
+  equal and trueEqual, one will check the data type while the other will check the value also. 
+  
 9. Why do front end frameworks exist?
+  
+  To take care of the tedious work of communicating with the server and providing with additional or cleaner functionality for various problems. 
+  
 
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
+  
+  It stores no information about you as a client
```

Comment:
and each request/response cycle is independent of those that came before and those that come after